util/stop_watch.h

```diff
@@ -101,36 +101,83 @@ class StopWatch {
   const uint64_t start_time_;
 };
 
+class SimpleStopWatch {
+ public:
+  bool IsStarted() { return start_ != 0; }
+  explicit SimpleStopWatch(SystemClock* clock) : clock_(clock), start_(0) {}
+  virtual void Start() = 0;
+  virtual uint64_t Elpased() = 0;
+  virtual ~SimpleStopWatch() = default;
+
+ protected:
+  SystemClock* clock_;
+  uint64_t start_;
+};
+
 // a nano second precision stopwatch
 class StopWatchNano {
  public:
-  explicit StopWatchNano(SystemClock* clock, bool auto_start = false)
-      : clock_(clock), start_(0) {
+  explicit StopWatchNano(SystemClock* clock, bool auto_start = false,
+                         bool cpu_time = false)
+      : clock_(clock), start_(0), is_time_type_cpu_(cpu_time) {
     if (auto_start) {
       Start();
     }
   }
-
-  void Start() { start_ = clock_->NowNanos(); }
-
+  void Start() {
+    if (is_time_type_cpu_) {
+      start_ = clock_->CPUNanos();
+    } else {
+      start_ = clock_->NowNanos();
+    }
+  }
   uint64_t ElapsedNanos(bool reset = false) {
-    auto now = clock_->NowNanos();
+    uint64_t now = 0;
+    if (is_time_type_cpu_) {
+      now = clock_->CPUNanos();
+    } else {
+      now = clock_->NowNanos();
+    }
     auto elapsed = now - start_;
     if (reset) {
       start_ = now;
     }
     return elapsed;
   }
-
   uint64_t ElapsedNanosSafe(bool reset = false) {
     return (clock_ != nullptr) ? ElapsedNanos(reset) : 0U;
   }
-
   bool IsStarted() { return start_ != 0; }
 
  private:
   SystemClock* clock_;
   uint64_t start_;
+  bool is_time_type_cpu_;
```

Comment:
Also (again) adding to the size of the object adds overhead.

You can make the class a template with a template parameter indicating whether to use CPU time or absolute time. You can then use something like `using StopWatchNano = Blah<blah>;' to preserve the original simple name.